docs/reference/docgen.md

```diff
@@ -0,0 +1,7 @@
+# Api reference generation
+
+The hardhat task `docgenMarkdown` is used to autogenerate the api documentation from solidity `natspec`.
+
+To generate the documentation locally run: `yarn hardhat docgenMarkdown`
```

Comment:
I think we should use hyphen for the command, like this one
https://hardhat.org/plugins/hardhat-deploy.html#_4-hardhat-etherscan-verify

Reply:
maybe `yarn hardhat generate-docs`

Reply:
:+1: